### PR TITLE
Implement use_latest_build_id for activities, child workflows, continue-as-new

### PR DESCRIPTION
**Pull request by dnr**
**What changed?**
Thread `use_latest_build_id` and build stamps where needed, respect `use_latest_build_id`.

**Why?**
Implement more feature

**How did you test it?**
Not yet

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by Sushisource — APPROVED** [review]
Makes sense!

File: common/util.go [hunk 1]
```diff
@@ -782,30 +783,41 @@ func OverrideWorkflowTaskTimeout(
 	return util.Min(taskStartToCloseTimeout, workflowRunTimeout)
 }
 
+// StampUsingVersioning returns the given WorkerVersionStamp if it is using versioning,
+// otherwise returns nil.
+func StampUsingVersioning(stamp *commonpb.WorkerVersionStamp) *commonpb.WorkerVersionStamp {
```
[the remark this answers is not in the document]
Comment (dnr) [reply common/util.go:788]: I tried for a while to come up with a good short name for this. `StampIfUsingVersioning`? `FilterStampUsingVersioning`? `StampOnlyUsingVersioning`?

I'm fine with yours too, though it sounds like it should be returning a bool not the stamp

File: service/history/workflow/mutable_state_impl.go [hunk 2]
```diff
@@ -2149,6 +2162,9 @@ func (ms *MutableStateImpl) ReplicateActivityTaskScheduledEvent(
 	scheduledEventID := event.GetEventId()
 	scheduleToCloseTimeout := attributes.GetScheduleToCloseTimeout()
 
+	// always use latest build id if activity is on a different task queue than the workflow
+	useLatestBuildId := attributes.UseLatestBuildId || attributes.TaskQueue.GetName() != ms.executionInfo.TaskQueue
```
[the remark this answers is not in the document]
Comment (dnr) [reply service/history/workflow/mutable_state_impl.go:2166]: can you elaborate? I'm not sure how what you said relates to the code

[the remark this answers is not in the document]
Comment (bergundy) [reply service/history/workflow/mutable_state_impl.go:2166]: Respect the SDK's decision and don't override it server side.

[the remark this answers is not in the document]
Comment (bergundy) [reply service/history/workflow/mutable_state_impl.go:2166]:
```
	// always use latest build id if activity is on a different task queue than the workflow
```
Should be:
Use the latest build ID if indicated by the SDK.

[the remark this answers is not in the document]
Comment (dnr) [reply service/history/workflow/mutable_state_impl.go:2166]: I got rid of all the task queue checks. I still have some concern about making the default behavior the least surprising thing, but it's up to the sdk now

**Pull request merged**